knife-supermarket.gemspec

```diff
@@ -14,5 +14,5 @@ Gem::Specification.new do |s|
   s.homepage    = 'https://github.com/cwebberOps/knife-supermarket'
   s.license       = 'Apache 2.0'
   s.require_paths = ["lib"]
-  s.add_dependency "chef", ">= 0.10.10"
+  s.add_dependency "chef", ">= 0.10.10", "<= 12.11.18"
```

Comment:
its probably better not to do this.  if you do this it just ensures that gem/bundler will not find this gem when 12.12.x chef is activated and will pick the prior version without any deprecation warnings.

its probably better to emit a deprecation warning on class parsing somewhere, and then have the classes NOP themselves if the chef_version is > 12.11.18 or whatever.